paddlenlp/transformers/gpt/modeling.py

```diff
@@ -195,11 +197,14 @@ def forward(self,
         # scale dot product attention
         product = layers.matmul(
             x=q, y=k, transpose_y=True, alpha=self.head_dim**-0.5)
+        key_length = product.shape[-2]
+        causal_mask = self.bias[:key_length, :key_length]
+        product = product * causal_mask
+        mask_score = (causal_mask - 1.0) * 10000.0
+        product = product + mask_score
         if attn_mask is not None:
-            # product = product + attn_mask
-            product = product * attn_mask
-            mask_score = (attn_mask - 1.0) * 10000.0
-            product = product + mask_score
+            product = product + attn_mask
+
```

Comment:
1. Why do we need `attn_mask` and `causal_mask` in the same time?
2. The `causal_mask` should be the sequence length, but it is initialized with `embed_dim`.  where `key_length` > `embed_dim`, there may be serious bugs.